### PR TITLE
[FLINK-23746][tests] Harden SuccessAfterNetworkBuffersFailureITCase.testSuccessfulProgramAfterFailure

### PR DESCRIPTION
**Pull request by tillrohrmann**
This commit decreases the parallelism used by the test job in order to reduce the time it takes to deploy
the whole job.

### CODE REVIEW

File: flink-tests/src/test/java/org/apache/flink/test/misc/SuccessAfterNetworkBuffersFailureITCase.java [hunk 1]
```diff
@@ -50,22 +50,22 @@
...
+                            .setNumberSlotsPerTaskManager(2)
                             .build());
 
     private static Configuration getConfiguration() {
         Configuration config = new Configuration();
-        config.set(TaskManagerOptions.MANAGED_MEMORY_SIZE, MemorySize.parse("80m"));
-        config.set(TaskManagerOptions.NETWORK_MEMORY_MIN, MemorySize.ofMebiBytes(32L));
-        config.set(TaskManagerOptions.NETWORK_MEMORY_MAX, MemorySize.ofMebiBytes(32L));
+        config.set(TaskManagerOptions.MANAGED_MEMORY_SIZE, MemorySize.parse("20m"));
+        config.set(TaskManagerOptions.NETWORK_MEMORY_MIN, MemorySize.ofMebiBytes(2L));
```
Comment (dmvk) [inline flink-tests/src/test/java/org/apache/flink/test/misc/SuccessAfterNetworkBuffersFailureITCase.java:67]: should this be also just divided by four?

**Review by dmvk — APPROVED** [review]
LGTM 👍 It doesn't seem necessary to have a higher parallelism for this test.

File: flink-tests/src/test/java/org/apache/flink/test/misc/SuccessAfterNetworkBuffersFailureITCase.java (hunk 1 above)
Comment (tillrohrmann) [reply flink-tests/src/test/java/org/apache/flink/test/misc/SuccessAfterNetworkBuffersFailureITCase.java:67]: Dividing by four would make the kmeans job pass. Hence, I needed to divide it by the 4^2.

**Comment by tillrohrmann** [comment]
Thanks for the review @dmvk. Merging once Azure gives green light.

**Comment by tillrohrmann** [comment]
Failing test seems to be unrelated. Merging this PR now.

**Pull request closed without merging**